scipy/stats/_stats_py.py

```diff
@@ -9691,9 +9785,59 @@ def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
     >>> wasserstein_distance([3.4, 3.9, 7.5, 7.8], [4.5, 1.4],
     ...                      [1.4, 0.9, 3.1, 7.2], [3.2, 3.5])
     4.0781331438047861
-
+    >>> wasserstein_distance([[0, 2, 3], [1, 2, 5]], [[3, 2, 3], [4, 2, 5]])
+    3.0
+    >>> wasserstein_distance([[0, 2.75], [2, 209.3], [0, 0]],
+    ...                      [[0.2, 0.322], [4.5, 25.1808]],
+    ...                      [0.4, 5.2, 0.114], [0.8, 1.5])
+    174.15840245217169
     """
-    return _cdf_distance(1, u_values, v_values, u_weights, v_weights)
+    m, n = len(u_values), len(v_values)
+    u_values = asarray(u_values).reshape(m, -1)
+    v_values = asarray(v_values).reshape(n, -1)
```

Comment:
I'd prefer that you just convert to arrays here, then next you just check whether both `ndim` are 1.

(After that, you can do all the rest of this stuff - getting `m`, `n` and ensuring that both arrays are 2d. I'd suggest raising an error if they have more than two dimensions.)

Reply:
> because the user could pass 2d arrays that actually could be computed with the 1d _cdf_distance , and which would be much faster

Yes, I know. I don't think it matters much either way.
* Option A (yours): faster in the unusual case that a user passes an `(n, 1)` array
* Option B (mine): *slightly* simpler (`ndim` instead of `shape[-1]` and no `ravel()`) and allows for the possibility of a unit test that compares optimization against `_cdf_distance`.

I considered both options before making the suggestion and thought that the benefits of B outweighed the benefits of option A. If you'd like to wait for someone else's opinion, we can do that, but I'd suggest that this is a minor point that should not hold up the PR. You can always submit a PR with optimizations like this after the initial PR merges; it would be a backward compatible change.

Reply:
Agreed:) I will keep it option B in this PR then.